### PR TITLE
Greeting message be1 implementation.

### PR DESCRIPTION
**Pull request by Robingoumaz**
Implementation of the lao#greet message according to the specification of [PR#917](https://github.com/dedis/popstellar/pull/917)

Changes :

- New messageData
- lao channel NewChannel returns an error
- hub now has a field for the server address
- the new message is computed and sent at the end of the newChannel function
- New tests and changes to old tests (to comply with the rest)

### CODE REVIEW

**Review by MaximeZmt — APPROVED** [review]
LGTM 👍 
Nice additions or the error to the NewChannel()

**Review by Warhedgehog — APPROVED** [review]
LGTM 👍

**Review by lhmerino — APPROVED** [review]
LGTM. Merge at will.

File: be1-go/channel/lao/mod.go [hunk 1]
```diff
@@ -132,7 +133,12 @@ func NewChannel(channelID string, hub channel.HubFunctionalities, msg message.Me
 
 	newChannel.registry = newChannel.NewLAORegistry()
 
-	return newChannel
+	err := newChannel.createAndSendLAOGreet()
+	if err != nil {
+		err = xerrors.Errorf("failed to send the greeting message: %v", err)
```
Comment (nkcr) [inline be1-go/channel/lao/mod.go:138]:
```suggestion
		return nil, xerrors.Errorf("failed to send the greeting message: %v", err)
```

File: be1-go/channel/lao/mod.go [hunk 2]
```diff
@@ -666,6 +672,67 @@ func compareLaoUpdateAndState(update messagedata.LaoUpdate, state messagedata.La
 	return nil
 }
 
+func (c *Channel) createAndSendLAOGreet() error {
```
Comment (nkcr) [inline be1-go/channel/lao/mod.go:675]: nice job, pleasant to read :)

**Review by nkcr — COMMENTED** [review]
Just a tiny remark, good job overall 👍

**Review by nkcr — APPROVED** [review]
🚀

**Pull request merged**